publish/publish.proj

```diff
@@ -185,8 +193,7 @@
   </Target>
   
   <Target Name="PublishDebToolPackageToFeed"
-          Condition="'$(PublishDebToolToFeed)' == 'true'">
-    <Error Condition="'$(CliNuGetFeedUrl)' ==''" Text="Missing required property CliNuGetFeedUrl" />
+          Condition="'$(PublishDebToolToFeed)' == 'true' AND '$(CliNuGetFeedUrl)' != ''">
```

Comment:
I strongly dislike having config plus a boolean that turns it on/off. I think it encourages people to leave sensitive info in the build but set up variables that are intended to disable using them, which leads to bad dev builds. With removing the info being the switch, if something bad happens (e.g. building a commit that doesn't support disabling the publish step), at worst you get a failed build.